tests/basctlServer.spec.ts

```diff
@@ -1,57 +1,56 @@
-import { assert, expect } from "chai";
-import * as sinon from "sinon";
+import { expect } from "chai";
+import { SinonSandbox, SinonMock, createSandbox, match } from "sinon";
 import * as net from 'net';
 import * as fs from 'fs';
 import { ActionJsonKey, ActionType } from "../src/actions/interfaces";
 import { mockVscode } from "./mockUtil";
 
 const testVscode = {
-    Uri: { parse: (value?: string, strict?: boolean) => {} },
+    Uri: {
+        parse: () => ""
+    },
     workspace: {
-        getConfiguration: () => {}
+        getConfiguration: () => ""
     },
     window: {
-        showErrorMessage: (message: string, ...items: string[]): Thenable<string | undefined> => {return Promise.resolve(undefined);}
+        showErrorMessage: (): Thenable<string | undefined> => Promise.resolve(undefined)
     }
 };
 
 const testSocket = {
-    on: (event: "data", listener: (data: Buffer) => void) => {},
-    write: (buffer: Uint8Array | string, cb?: (err?: Error) => void) => true
-}
+    on: () => "",
+    write: () => true
+};
 
 const testServer = {
-    listen: (path: string, backlog?: number, listeningListener?: () => void) => {},
-    close: (callback?: (err?: Error) => void) => {}
-}
+    listen: () => "",
+    close: () => ""
+};
 
 mockVscode(testVscode, "src/actions/performer.ts");
 mockVscode(testVscode, "src/actions/actionsFactory.ts");
 import * as performer from '../src/actions/performer';
 import * as actionsFactory from '../src/actions/actionsFactory';
 import { closeBasctlServer, startBasctlServer } from "../src/basctlServer/basctlServer";
-import { Stream } from "stream";
 
 
 describe("basctlServer", () => {
-    let sandbox: any;
-    let performerMock: any;
-    let actionsFactoryMock: any;
-    let windowMock: any;
-    let fsMock: any;
-    let netMock: any;
+    let sandbox: SinonSandbox;
+    let performerMock: SinonMock;
+    let actionsFactoryMock: SinonMock;
+    let windowMock: SinonMock;
+    let fsMock: SinonMock;
+    let netMock: SinonMock;
     let socketMock: any;
     let serverMock: any;
```

Comment:
```suggestion
    let socketMock: SinonMock;
    let serverMock: SinonMock;
```

Reply:
It will not work here with SinonMock, do not know why

Reply:
Done